actioncable/lib/action_cable/storage_adapter/base.rb

```diff
@@ -0,0 +1,22 @@
+module ActionCable
+  module StorageAdapter
+    class Base
+      attr_reader :logger, :server
+
+      def initialize(server)
+        @server = server
+        @logger = @server.logger
+      end
+
+      # Storage connection instance used for broadcasting. Not intended for direct user use.
+      def broadcast
+        raise NotImplementedError
+      end
```

Comment:
`broadcast` feels off, as it just returns a dedicated `pubsub` for broadcasting. Would do:

```ruby
def publish_broadcast(*args) # :nodoc:
  broadcaster.publish(*args)
end
```

Note: adding `# :nodoc:` means we don't have to write that it's not intended for direct user use.